### PR TITLE
fix(mongo back): fix append & join to handle domain step as str [TCTC-1829]

### PR DESCRIPTION
**Pull request by raphaelvignes**
Append & Join steps of mongo back translator. should handle the domain name provided as a str directly

### CODE REVIEW

File: server/tests/backends/fixtures/append/domain_name_only.json [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        "AGE": 43,
+        "SCORE": 200
+      },
+      {
+        "NAME": "miam",
+        "AGE": 999,
+        "SCORE": 12
+      }
+    ]
+  }
```
Comment (austil) [inline server/tests/backends/fixtures/append/domain_name_only.json:153]: Is this used or not ?

File: server/tests/backends/fixtures/append/domain_name_only.json [hunk 2]
```diff
@@ -0,0 +1,154 @@
+{
+  "exclude": [
+    "snowflake",
+    "mysql",
+    "postgres",
+    "pandas"
+  ],
```
Comment (austil) [inline server/tests/backends/fixtures/append/domain_name_only.json:7]: Should we also run this test for `pandas` & `snowflake` ?

**Review by austil — APPROVED** [review]
Fix looks good :green_circle: 
Some questions on tests :wink:

File: server/tests/backends/fixtures/append/domain_name_only.json (hunk 2 above)
Comment (raphaelvignes) [reply server/tests/backends/fixtures/append/domain_name_only.json:7]: let's see :D

Comment (raphaelvignes) [reply server/tests/backends/fixtures/append/domain_name_only.json:7]: :heavy_check_mark: for snowflake

File: server/tests/backends/fixtures/append/domain_name_only.json (hunk 1 above)
Comment (raphaelvignes) [reply server/tests/backends/fixtures/append/domain_name_only.json:153]: yes for Snowflake as result differs from Snowflake vs Pandas

**Pull request merged**